### PR TITLE
Media Extractor / Media Summary: move with other libraries.

### LINKED ISSUE #47: Relocate the `Media Extractor` and `Media Summary` classes
We're using `jetpack_require_lib()` from WPCOM since https://github.com/Automattic/jetpack/commit/28fdbcd9626ad2393bcea898f223285ff58bcadb four months ago -- would it be wiser to migrate `class.media-extractor.php` and `class.media-summary.php` into the `_inc/lib` folder and use jetpack_require_lib() to pull them in, rather than have them globally included in the root?

cc: @justinshreve

### PR DESCRIPTION
**Pull request by jeherve**
Replaces #977
Fixes #47

### CODE REVIEW

**Comment by eliorivero** [comment]
It's good. It just needs a rebase.
When this is applied, the build script will have to be updated to match the new location of the libraries.

**Comment by jeherve** [comment]
Rebased.

File: tests/php/_inc/lib/test_class.jetpack-media-summary.php [hunk 1]
```diff
@@ -1,5 +1,9 @@
 <?php
 
+if ( ! class_exists( 'Jetpack_Media_Summary' ) ) {
+			jetpack_require_lib( 'class.media-summary' );
```
Comment (zinigor) [inline tests/php/_inc/lib/test_class.jetpack-media-summary.php]: It looks like there's something wrong with indentation here.

**Comment by zinigor** [comment]
Please do not merge until 3.9.2 hits.

File: tests/php/_inc/lib/test_class.jetpack-media-summary.php (hunk 1 above)
Comment (jeherve) [inline tests/php/_inc/lib/test_class.jetpack-media-summary.php]: Fixed.

**Comment by kraftbj** [comment]
@zinigor Safe to merge or still waiting for 3.9.3?

**Comment by zinigor** [comment]
No, please wait until 3.9.3 is in its branch. This probably can be moved to 3.10, by the way.

**Pull request merged**